lib/buffer.js

```diff
@@ -208,12 +214,14 @@ function assertSize(size) {
   let err = null;
 
   if (typeof size !== 'number') {
-    err = new TypeError('"size" argument must be a number');
+    // err = new TypeError('"size" argument must be a number');
+    err = new errors.TypeError('ERR_INVALID_ARG_TYPE', 'size', 'number', size);
   } else if (size < 0) {
-    err = new RangeError('"size" argument must not be negative');
+    err = new errors.RangeError('ERR_VALUE_OUT_OF_RANGE',
+      '"size" argument must not be negative');
   } else if (size > binding.kMaxLength) {
-    err = new RangeError('"size" argument must not be larger ' +
-                         'than ' + binding.kMaxLength);
+    err = new errors.RangeError('ERR_VALUE_OUT_OF_RANGE',
```

Comment:
This is probably a good candidate for adding a function that takes info about the arguments and the expected range as opposed to using  a fixed string.